src/Post/Post.js

```diff
@@ -364,76 +364,88 @@ function CommentSection({
     </section>
   );
 }
-function MediaSection({ media_items, user_name }) {
-  const [mediaIndex, setMediaIndex] = React.useState(0);
+function width(el) {
+  return el?.getBoundingClientRect().width;
+}
+function MediaSection({media_items}) {
+  const mediaContainer = React.useRef();
+  const mediaContainerWidth = width(mediaContainer.current);
+  const [containerWidth, setContainerWidth] = React.useState(0);
+  const [scrollLeft, setScrollLeft] = React.useState(0);
   const [videoPause, setVideoPause] = React.useState(false);
-  return (
-    <section className="img-container">
-      {media_items[mediaIndex].type === 'photo' ? (
-        <img
-          className="post-img"
-          src={media_items[mediaIndex].url}
-          alt={`${user_name}  Avatar`}
-          loading="lazy"
-        />
-      ) : (
-        <>
-          {videoPause && (
-            <div className="play-icon-container">
-              <PlayArrowRounded
-                onClick={() => setVideoPause(false)}
-                className="play-icon"
-                fontSize="large"
-              />
-            </div>
-          )}
+  const scrollLimit = -1 * (mediaContainerWidth - containerWidth);
 
-          <div
-            onClick={() => setVideoPause(!videoPause)}
-            className="post-video-container"
-          >
-            <Video
-              videoURL={media_items[mediaIndex].url}
-              paused={videoPause}
-              className="post-video"
-              autoPlay={false}
-            />
+  const [mediaIndex, setMediaIndex] = React.useState(0);
+  return (
+    <section
+      className="media-section"
+      ref={(ref) => setContainerWidth(width(ref))}
+    >
+      <div
+        className="media-container"
+        ref={mediaContainer}
+        style={{ transform: `translateX(${scrollLeft}px)` }}
+      >
+        {media_items?.map((mediaItem) => (
+          <div>
+            {mediaItem.type === 'photo' ? (
+              <PostImage imageURL={mediaItem.url}/>
+            ) : (
+              <div
+                onClick={() => setVideoPause(!videoPause)}
+                className="post-video-container"
+              >
+                <PostVideo
+                  videoURL={mediaItem.url}
+                  paused={videoPause}
```

Comment:
This means every video in the post will pause and unpause at the same time. I think the pausing state should be inside `PostVideo`. This is a tricky one. We can do it together.

Reply:
I don't think that's the case here because if you see the real IG when you press the next button or the previous button when the video is playing, the video pauses, which means the buttons need access to the state in order to pause the video.

Reply:
That's correct! But they need access to a specific video, this prop will change for every video.

Reply:
On it